File: CS/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using System.Windows.Input;

namespace Example.ViewModel {
    public class MainViewModel : ISupportServices {
        IServiceContainer serviceContainer = null;
        protected IServiceContainer ServiceContainer {
            get {
                if(serviceContainer == null)
                    serviceContainer = new ServiceContainer(this);
                return serviceContainer; 
            }
        }
        IServiceContainer ISupportServices.ServiceContainer { get { return ServiceContainer; } }
        IMessageBoxService MessageBoxService { get { return ServiceContainer.GetService<IMessageBoxService>(); } }

        public ICommand ShowMessageCommand { get; private set; }
        public MainViewModel() {
            ShowMessageCommand = new DelegateCommand(ShowMessage);
        }
        void ShowMessage() {
            MessageBoxService.Show("This is MainView!");
        }
    }
}

File: VB/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System.Windows.Input

Namespace Example.ViewModel

    Public Class MainViewModel
        Implements ISupportServices

        Private _ShowMessageCommand As ICommand

        Private serviceContainerField As IServiceContainer = Nothing

        Protected ReadOnly Property ServiceContainerProp As IServiceContainer
            Get
                If serviceContainerField Is Nothing Then serviceContainerField = New ServiceContainer(Me)
                Return serviceContainerField
            End Get
        End Property

        Private ReadOnly Property ServiceContainer As IServiceContainer Implements ISupportServices.ServiceContainer
            Get
                Return ServiceContainerProp
            End Get
        End Property

        Private ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return ServiceContainerProp.GetService(Of IMessageBoxService)()
            End Get
        End Property

        Public Property ShowMessageCommand As ICommand
            Get
                Return _ShowMessageCommand
            End Get

            Private Set(ByVal value As ICommand)
                _ShowMessageCommand = value
            End Set
        End Property

        Public Sub New()
            ShowMessageCommand = New DelegateCommand(AddressOf ShowMessage)
        End Sub

        Private Sub ShowMessage()
            MessageBoxService.Show("This is MainView!")
        End Sub
    End Class
End Namespace
